File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/services/HouseService.java
package com.chaincuerealestate.chaincuerealestate.services;

import com.chaincuerealestate.chaincuerealestate.domains.House;
import com.chaincuerealestate.chaincuerealestate.exceptions.HouseNotFoundException;
import com.chaincuerealestate.chaincuerealestate.repositories.HouseRepository;
import com.chaincuerealestate.chaincuerealestate.utilities.AweS3Urls;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class HouseService implements HouseServiceI {
    private final HouseRepository houseRepository;

    @Override
    public House save(House.HouseTypes houseTypes) {
        var houseImage = House.create(houseTypes, AweS3Urls.URLFrontImage1);
        return houseRepository.save(houseImage);
    }

    @Override
    public House findById(String id) {
        return houseRepository.findById(id).orElseThrow(() -> new HouseNotFoundException(id));
    }

    @Override
    public List<House> findAll() {
        return houseRepository.findAll();
    }

}

File: kotlin/chaincue-real-estate/src/main/kotlin/com/example/chaincuerealestate/services/HouseService.kt
package com.example.chaincuerealestate.services

import com.example.chaincuerealestate.domains.House
import com.example.chaincuerealestate.exceptions.HouseNotFoundException
import com.example.chaincuerealestate.repositories.HouseRepository
import com.example.chaincuerealestate.utilities.AweS3Urls
import org.springframework.stereotype.Service

@Service
class HouseService(private val houseRepository: HouseRepository) : HouseServiceI {

    override fun save(houseTypes: House.HouseTypes): House {
        val house = House.create(houseTypes, AweS3Urls.URLFrontImage1)
        return houseRepository.save(house)
    }

    override fun findById(id: String): House {
        return houseRepository.findById(id)
            .orElseThrow { HouseNotFoundException(id) }
    }

    override fun findAll(): List<House> {
        return houseRepository.findAll()
    }
}

